### DISCUSSION TITLE
Is there a way to get current stack-frames or call-stack at a certain point ?

### DISCUSSION THREAD

**Discussion by abdelhamid-f-nasser** [facebook/hermes#1261] [Q&A]
I'm trying to get the current point of the call-stack to help with debugging and performance. What I am trying to do might be similar to how the debugger allows us to pause the JS Thread and check the current call stack.

_Note: Please pardon me if this is not the proper place to ask, I'm not sure which repo I should drop this question._

**Comment by fbmal7** [comment]
You can construct an error object and inspect its `stack` property.

```
function foo() {
  var stack_str = new Error().stack;
  print(stack_str);
}

foo();
```

This prints the following:

```
Error
    at foo (file.js:2:28)
    at global (file.js:6:4)
```

**Reply by abdelhamid-f-nasser** [reply]
Thank you for your prompt response, @fbmal7 🙏 ! Is there any I can use the above approach to mimic the developer tools, where it pauses or interrupt the execution and retrieves the current call stack ?
